source/common/http/conn_manager_impl.cc

```diff
@@ -1010,6 +1032,7 @@ void ConnectionManagerImpl::ActiveStream::encode100ContinueHeaders(
 void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilter* filter,
                                                         HeaderMap& headers, bool end_stream) {
   resetIdleTimer();
+  disarmRequestTimer();
```

Comment:
This is only needed in case a local reply is made, because in the event of an upstream request the callback has already caused the request time to be disarmed, right? (I'm just making sure I understand.)